src/linux_platform.cpp

```diff
@@ -519,4 +592,326 @@ void LinuxPlatform::cmdLineArgs(int argc, char** argv)
     memcpy(_args, argv, argc * sizeof(char*));
     _args[argc] = 0;
 }
-#endif
\ No newline at end of file
+
+void LinuxPlatform::setupGpio(uint32_t dwPin, uint32_t dwMode)

```

Comment:
Maybe wiringpi or something similar is a bit faster? (You mentioned timing problems in the readme.)

Reply:
I thought about WiringPI. Then I came accross this bad news: http://wiringpi.com/wiringpi-deprecated/
Bottom line: the author is fed up.

So should we still go with this? 
WiringPi offers both: direct HW access and sysfs.

For GPIO and SPI I use the standard kernel interface without any additional library at the moment. The SPI stuff here was inspired by WiringPi anyway. ;)
The current sysfs stuff is currently not used in proper way. I always open/close the files which is horribly slow. So maybe by keeping it open, it may work a low better.

Using the standard kernel interfaces would make it easy to use other boards (not RPi) as well.

But maybe it is really better to go with WiringPi although the orignal author has abandonned it. I am pretty sure that someone will take over this.

What do you think?

Reply:
Maybe we should leave the Linux platform as generic one as is for KNX IP only and introduce a new platform specifically for Raspberry Pi.
Then I could implement the direct GPIO HW access there.
For SPI I would use the SPI kernel interface as I do now.

Reply:
Maybe we should leave the Linux platform as generic one as is for KNX IP only and introduce a new platform specifically for Raspberry Pi.
Then I could implement the direct GPIO HW access there.
For SPI I would use the SPI kernel interface as I do now.

Reply:
I think standard linux interfaces would be better. So the actual linux board doesn't matter. 
You can keep it as it is. Maybe you or I or somebody else will improve it later.

Reply:
done.